src/python/pants/backend/awslambda/python/target_types.py

```diff
@@ -31,8 +31,8 @@ class PythonAwsLambdaRuntime(StringField):
     value: str
 
     @classmethod
-    def compute_value(cls, raw_value: Optional[str], *, address: Address) -> str:
-        value = cast(str, super().compute_value(raw_value, address=address))
+    def sanitize_raw_value(cls, raw_value: Optional[str], *, address: Address) -> str:
+        value = cast(str, super().sanitize_raw_value(raw_value, address=address))
```

Comment:
I agree with using the same method for both types of `Field`s. I originally wanted to separate them to emphasize how the two APIs are consumed differently, but that's not worth it. It makes it confusing to navigate between both.

I'm not sure which name I prefer, though: `compute_value` vs. `sanitize_raw_value`. I'm conflicted because:

1. I think `sanitize_raw_value` is more descriptive, but
2. Several plugin authors are already using `compute_value` for their target definitions.

We want to minimize churn. But we should also do what's going to be best in the long term; better to have churn now before even more people write plugins.

Thoughts?

Reply:
Hm, great point. I think `.compute_value()` seems more natural, I was more worried that people might infer it to mean "it's ok to do heavy computation in this method" -- the docstring clarifies that a lot, though. If people are already using it in plugins, we might consider adding `@deprecated` to `.sanitize_raw_value()`?

Reply:
If we go with `.compute_value()`, then I'm not as worried about us needing to deprecate `.sanitize_raw_value()`. I will be really shocked if people subclassed `AsyncField` _and_ overrode `.sanitize_raw_value()`. We don't document how to do that, for example. And it's rare to use an `AsyncField`.

We're also allowed to make breaking changes without deprecation, so long as we document at https://www.pantsbuild.org/docs/plugin-upgrade-guide, but in good faith, we want to minimize churn as much as possible.

Reply:
That sounds great! Will do!